### ISSUE TITLE
Imported distro doesn't run with a default user and interim write permission issues in /mnt/c/Users/$me/ folders

### ISSUE THREAD

**Issue by delphym** [microsoft/WSL#4268]
## ENVIRONMENT
Running W10 Pro, v1903, OS build: 18362.17

## Background
Originally in the past I've installed an Ubuntu WSL distro from the store, went through the all necessary setup as described [here](https://docs.microsoft.com/en-us/windows/wsl/initialize-distro).  Kept it up-to-date, so as of now it reports as Ubuntu 16.04.6 LTS Xenial.

## Reproduction
With the upgrade of W10 to 1903 I backed up my WSL distro with new --exporte <distro> feature and lucky me, a few days later I did something bad to my distro, and sort of killed it, so pretty much I had to --terminate and --unregister it. Followed by --import of my backup WSL.

## Results
Well, it worked OK, except when I open Ubuntu from a Start menu (as usually) it opens Ubuntu console under the `root` user only and I had to use `login $me` or `sudo su - $me` to get my shell (zsh actually). 

### Other observations
```
C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps>dir /aL
 Volume in drive C is Windows
 Volume Serial Number is CAE9-0D87

 Directory of C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps

04/06/2019  13:19                 0 MicrosoftEdge.exe
07/06/2019  19:18                 0 python.exe
07/06/2019  19:18                 0 python3.exe
23/08/2018  14:19                 0 ubuntu.exe
               4 File(s)              0 bytes
               0 Dir(s)  251,030,593,536 bytes free
```
Just normal listing
```
C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps>dir
 Volume in drive C is Windows
 Volume Serial Number is CAE9-0D87

Directory of C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps

10/04/2018  10:17    <DIR>          46932SUSE.openSUSELeap42.2_022rs5jcyhyac
30/05/2019  10:51    <DIR>          Backup
18/10/2018  07:55    <DIR>          CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc
07/06/2019  19:18    <DIR>          Microsoft.DesktopAppInstaller_8wekyb3d8bbwe
04/06/2019  13:19    <DIR>          Microsoft.MicrosoftEdge_8wekyb3d8bbwe
04/06/2019  13:19                 0 MicrosoftEdge.exe
07/06/2019  19:18                 0 python.exe
07/06/2019  19:18                 0 python3.exe
23/08/2018  14:19                 0 ubuntu.exe
               4 File(s)              0 bytes
               5 Dir(s)  251,020,718,080 bytes free
```
Some other attempts:
```
C:\Users\dmladek>ubuntu.exe config --default-user delphym
The system cannot find the file C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps\ubuntu.exe.

C:\Users\dmladek>dir C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps\ubuntu.exe
 Volume in drive C is Windows
 Volume Serial Number is CAE9-0D87

 Directory of C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps

23/08/2018  14:19                 0 ubuntu.exe
               1 File(s)              0 bytes
               0 Dir(s)  252,973,445,120 bytes free

C:\Users\dmladek>lxrun
'lxrun' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\dmladek>ubuntu.exe /?
The system cannot find the file C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps\ubuntu.exe.
```
And the last but one
```
C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps>ubuntu
The system cannot find the file C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps\ubuntu.exe.

C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps>cd CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc

C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps\CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc>dir
 Volume in drive C is Windows
 Volume Serial Number is CAE9-0D87

 Directory of C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps\CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc

18/10/2018  07:55    <DIR>          .
18/10/2018  07:55    <DIR>          ..
30/05/2019  10:51                 0 ubuntu.exe
               1 File(s)              0 bytes
               2 Dir(s)  251,024,666,624 bytes free

C:\Users\dmladek\AppData\Local\Microsoft\WindowsApps\CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc>ubuntu.exe
root@DMLADEK-NZ-LE:~#
```
Another `ubuntu.exe` file, but not zero size!
```
c:\Program Files\WindowsApps\CanonicalGroupLimited.UbuntuonWindows_1804.2019.521.0_x64__79rhkp1fndgsc>dir
 Volume in drive C is Windows
 Volume Serial Number is CAE9-0D87

 Directory of c:\Program Files\WindowsApps\CanonicalGroupLimited.UbuntuonWindows_1804.2019.521.0_x64__79rhkp1fndgsc

30/05/2019  10:47    <DIR>          .
30/05/2019  10:47    <DIR>          ..
30/05/2019  10:47           219,034 AppxBlockMap.xml
30/05/2019  10:47             3,790 AppxManifest.xml
30/05/2019  10:46    <DIR>          AppxMetadata
30/05/2019  10:47            11,212 AppxSignature.p7x
30/05/2019  10:46    <DIR>          Assets
30/05/2019  10:47       231,179,584 install.tar.gz
27/05/2018  07:49             5,384 resources.pri
30/05/2019  10:47           211,968 ubuntu.exe
               6 File(s)    231,630,972 bytes
               4 Dir(s)  251,023,949,824 bytes free

c:\Program Files\WindowsApps\CanonicalGroupLimited.UbuntuonWindows_1804.2019.521.0_x64__79rhkp1fndgsc>ubuntu
Access is denied.

c:\Program Files\WindowsApps\CanonicalGroupLimited.UbuntuonWindows_1804.2019.521.0_x64__79rhkp1fndgsc>
```

## Expected Results
* When a new Ubuntu console is opened, then it would login me with the default user (well defined in the past)
* Import function would have parameter to specify the default user
* There will be any other documented mean how to achieve to specify default user for the imported distro.

## Other experiments/attempts
I was not happy with this state and I also imported it into a different directory when previously installed WSL resided. So, I unregistered the distro,  try to install it from the store, then unregister just the installed one, and imported my backup to the location where the installed WSL (the later one) was installed. 

However still the same results - starting it as `root`

## Other 'r/w' access issues - non reproducible issue
Since I unregistered my originally installed WSL, which I was using also to `git checkout` code into the Windows space, I mean e.g. into `/mnt/c/Users/$me/SRC/sharedGITfolder_for_Windows_and_Linux_Access`
I sometimes don't have `r/w` access anywhere outside the WSL space (it's root FS).
I cannot figure out what is the cause, but it must be the way how I'm invoking my Ubuntu WSL.

Since those issues I open Ubuntu console in following ways:
* via Start menu and typing Ubuntu and hitting Enter key on the 'Ubuntu App'
** typing `login $me`
** typing `sudo su -u $me`
* via Start menu typing `cmd` followed by Enter
** typing `wsl -u $me`
* via Start menu typing `cmd` and running it as Admin. (this is very rare)
** typing `wsl -u $me`

**Comment by delphym** [comment]
I found this issue: [3974](https://github.com/microsoft/WSL/issues/3974), and modifying the registry key
`HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Lxss\{MY-UUID}` as suggested, it resolved my main problem, but still the randomness of those r/w, respectively r/o and the strange mess with missing ubuntu.exe  still persist.

**Comment by therealkenc** [comment]
> respectively r/o and the strange mess with missing ubuntu.exe still persist.

Yes, true that. That is #3974 (message) "...to the extent `--export` isn't bundling up (and `--import` isn't restoring) everything PackageFamilyName related including the launcher."

Which, it isn't.

If you want a short answer: `wsl.exe --import` isn't, in its present form, especially useful for anything related restoring your Store app and the related registry entries. Once it is gone the only _practical_ way to get your `ubuntu.exe` back is to reinstall it, and pretend you don't know those very long path names under `c:\Program Files\WindowsApps\` exist. [There are impractical solutions to everything, mind.]

Ref #3974 #3536 #2955

The problem (if that is the word) isn't limited to WSL. There isn't a really good way to restore your Candy Crush Saga [either](https://answers.microsoft.com/en-us/windows/forum/windows_10-windows_store-winpc/how-to-backup-apps-and-games-installed-from-the/2e7d2bda-6209-4f48-b37f-b3d7f7e9ce3d?auth=1&page=1). The whole Android, iPhone (and if were still a thing) Windows Phone App Store model is based on a redownload/reinstall.

**Comment by delphym** [comment]
Thanks very much for your answers...

Yes, I could see that `wsl.exe --import` is not very useful with restoring the necessary settings around that particular disto. Also, it is not clear that default user, etc. is somehow external to any WSL distro instance.

OKi, so what would be "right" way to imitate the setup process which you'll get when you install your favourite distro from the store?

Say, you have a clean PC, just enabled WSL on it, and on an external medium you have a TAR with a ditro you would like to use it on your clean PC, BUT, in the same convenient way as it was working on the other computer from where did you had it exported?

**Comment by therealkenc** [comment]
Well, there isn't really anything to "imitate" -- I am not even sure what imitation would look like versus the real deal. Since, presumably, you'd want to get as close to the real deal as possible with your imitation. And getting close isn't necessary. The gory details of the launcher is [here](https://github.com/microsoft/WSL-DistroLauncher) and there is some information on side-loading [here](https://devblogs.microsoft.com/commandline/wsl-distro-launcher/) and manual installation [here](https://docs.microsoft.com/en-us/windows/wsl/install-manual). But armed with all that knowledge, you can't do anything you can't do by clicking the 'install' button from the store.

Do a fresh install. Take that tarball and copy it into the new running WSL instance of the same pedigree (say Ubuntu) as the original. Then `tar xfp` the thing onto `/` as `root`. Like [this](https://help.ubuntu.com/community/BackupYourSystem/TAR).

Not that that is how I'd roll personally. Extract just /home and then `apt get` whatever you fancy. Your old tarball is full of cruft you apt-got and haven't used lately anyway. . ;)

**Comment by therealkenc** [comment]
Consolidating into (now closed) #3974. The newspeak is to use `/etc/wsl.conf` `[user] default=username`, and have that set up either before the `--export` or fix it up after the `--import`.

**Comment by imadelq** [comment]
It worked for me !
I ve imported wsl Ubuntu-20.04 with a new name **Ubuntu** in D: drive
If you have another name for example "MyUbuntu", change it to Ubuntu
In order to add ubuntu to c:\Program Files\WindowsApps\, I have downloaded Ubuntu ( not Ubuntu-20.04 !! ) using the following command :
**wsl --install -d Ubuntu**
I just want to install new Ubnutu, When Ubuntu was downloaded, surprise, It runs with my imported version, because they have the same name !
Now if you want to logging with user name use the following command
**ubuntu.exe config --default-user yourUserName**
